doc/catalyst-cli/__init__.rst

```diff
@@ -0,0 +1,42 @@
+Catalyst CLI
+============
+
+Catalyst includes a standalone command-line-interface compiler tool ``catalyst-cli`` that quantum
+compiles MLIR input files into an object file, independent of the Python frontend.
+
+This compiler tool combines three stages of compilation:
+
+#. ``quantum-opt``: Performs the MLIR-level optimizations and lowers the input dialect to the LLVM dialect.
+#. ``mlir-translate``: Translates the input in the LLVM dialect into LLVM IR.
+#. ``llc``: Performs lower-level optimizations and creates the object file.
+
+``catalyst-cli`` runs all three stages under the hood by default, but it also has the ability to run
+each stage individually. For example:
+
+.. code-block:: console
+
+    # Creates both the optimized IR and an object file
+    catalyst-cli input.mlir -o output.o
+
+    # Only performs MLIR optimizations
+    catalyst-cli --tool=opt input.mlir -o llvm-dialect.mlir
```

Comment:
Can we have a more sophisticated example of `opt`, where some passes and post-mortem are actually performed? e.g. https://github.com/PennyLaneAI/catalyst/blob/main/mlir/test/cli/DumpBeforeAfterPass.mlir#L15

Reply:
I've added an example, do you mind taking a look?